### DISCUSSION TITLE
Any hope of filtering (specifically for exclusion) for project linters?

### DISCUSSION THREAD

**Discussion by jantmanAtCox** [oxsecurity/megalinter#3407] [Q&A]
Greetings! I'm on a code quality crusade at my day job, and MegaLinter seems absolutely amazing, so thanks so much for this project!

One issue we're having with a mass rollout, though, is the inability to exclude files or directories globally, i.e. for every enabled linter (many of which are running in project mode). I was wondering if there are any plans, or any hope, of a functionality for globally excluding certain paths?

We have a polyglot environment with many projects in various stages of lifecycle, and it would really help adoption of MegaLinter if there was a quick way (i.e. by environment variable) of completely excluding certain paths from every linter, without having to manage configuration files for each of those linters.

**Comment by Jayllyz** [comment]
You can use`EXCLUDED_DIRECTORIES` or `FILTER_REGEX_EXCLUDE` in your `.mega-linter.yml` file. (https://megalinter.io/latest/config-variables/)

**Reply by jantmanAtCox** [reply]
Oh cool, thanks. Maybe this is processed differently if it's in the YML file than if it's an environment variable? I tried it as an environment variable, but it did nothing for the large number of linters that were running in `project` mode.

**Reply by Jayllyz** [reply]
If you used Filter regex, it is not applicable with linters using CLI lint mode project
-> https://megalinter.io/latest/config-filtering/

**Reply by nvuillam** [reply]
When the list of files is sent as parameter to the linter ( CLI lint mode = list_of_files or file), that's easy to filter the list of file sent by MegaLinter

But when the linter is called in project mode on the root of the repo, we would need to convert the MegaLinter config inclusion/exclusion config into each linter's own format (which is different for all linters: can be a CLI arguent, a config file property, a .xxxignore file ...)

That's why we can not generate them "on the fly"... and they have to be written manually by the analyzed repository maintainers :/

**Reply by jantmanAtCox** [reply]
> That's why we can not generate them "on the fly"... and they have to be written manually by the analyzed repository maintainers :/

I guess this is what I was getting at with my question. The documentation for many (but certainly not all) of the linters says that if their configuration file isn't found, one from https://github.com/oxsecurity/megalinter/tree/main/TEMPLATES will be used. What are the odds of an enhancement, where those template config files could be dynamically generated to include a handful of the most common options, at least file exclusions?

**Reply by nvuillam** [reply]
@jantmanAtCox it would take days of work... if someone finds a way to industrialize that, probably with descriptor properties, I would be ok to accept the PR, and the linters could be "ignore-managed" incrementally...
But it would require something like `REGEX to .gitignore` format... do you know how to do that ?

**Reply by nvuillam** [reply]
Another option could be to implement the management of some .megalinterignore file, it would probably be easier to convert into the required linter formats
